{"cells": [{"cell_type": "code", "execution_count": 1, "id": "889dd944", "metadata": {}, "outputs": [], "source": ["# This Python 3 environment comes with many helpful analytics libraries installed\n# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python\n# For example, here's several helpful packages to load\n\nimport numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\nfrom pandas_profiling import ProfileReport #pandas profiling tool\n\n# Input data files are available in the read-only \"../input/\" directory\n# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n\nimport os\nfor dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        print(os.path.join(dirname, filename))\n\n# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\" \n# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session"]}, {"cell_type": "code", "execution_count": 1, "id": "99324bdc", "metadata": {}, "outputs": [], "source": ["desc = pd.read_csv(\"../input/mygenome/genome_file_description.csv\")\ngenome_df = pd.read_csv(\"../input/mygenome/genome_zeeshan_usmani.csv\")\ndesc"]}, {"cell_type": "code", "execution_count": 1, "id": "d95f2d2b", "metadata": {}, "outputs": [], "source": ["genome_df"]}, {"cell_type": "code", "execution_count": 1, "id": "133fe14b", "metadata": {}, "outputs": [], "source": ["Genome = ProfileReport(genome_df)\nGenome"]}, {"cell_type": "code", "execution_count": 1, "id": "9ee887f5", "metadata": {}, "outputs": [], "source": ["genome_df.chromosome.value_counts().to_frame().rename(columns={0:\"Unique Values\"}).style.background_gradient(cmap=\"plasma\")"]}, {"cell_type": "code", "execution_count": 1, "id": "8624b7f0", "metadata": {}, "outputs": [], "source": ["genome_df.genotype.value_counts().to_frame().rename(columns={0:\"Unique Values\"}).style.background_gradient(cmap=\"plasma\")"]}, {"cell_type": "markdown", "id": "e54688e3", "metadata": {}, "source": ["Work in progress\n\n* Accessing SNPedia api to analye on the basis of rsid and genotype\n\n### Next Steps\n\n* Create a method to automatically generate report"]}], "metadata": {"kernelspec": {"display_name": "Python 3 (ipykernel)", "language": "python", "name": "python3"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.9.9"}}, "nbformat": 4, "nbformat_minor": 5}